src/core.py

```diff
@@ -156,39 +202,49 @@ def adapt_connections(
     Args:
         graph: The FX graph
         new_node: The node whose connections need adaptation
-        new_node_shape: The shape of the new node
+        new_node_shape: The shape of the new node, can be None if the new node is a skip connection
         input_shape: The shape of the input node (pre-computed)
         output_shape: The shape of the output node (pre-computed)
     Returns:
         graph: The modified graph
     """
     
-    # check if the input node output shapes are compatible with the node to adapt from
-    if input_shape is not None:
+    # Special handling for skip connections (torch.add operations)
+    if new_node.target == torch.add:
+        # Get shapes of both input nodes
+        first_node = new_node.args[0]
+        second_node = new_node.args[1]
+        first_shape = first_node.meta['tensor_meta'].shape
+        second_shape = second_node.meta['tensor_meta'].shape
+        
+        # If shapes don't match, adapt both inputs to the larger shape
+        if first_shape[-1] != second_shape[-1]:
+            target_size = max(first_shape[-1], second_shape[-1])
+            
+            # Adapt first node if needed
+            if first_shape[-1] != target_size:
+                graph, first_node = adapt_node_shape(graph, first_node, first_shape[-1], target_size)
+            
+            # Adapt second node if needed
+            if second_shape[-1] != target_size:
+                graph, second_node = adapt_node_shape(graph, second_node, second_shape[-1], target_size)
```

Comment:
I think this is the same logic as in utils.py branch function, should we abstract it to a function? Not sure if the next two lines can also be included (which would let you include the if statement too).

Reply:
I agree with you, fixed in core.py. We'll leave the code in just in case we change our minds about initializing them with the same shape, but as it stands the adaption checks will always fail and be ignored.

Reply:
I think you commented on the wrong comment here, but cool sounds good. First comment here still stands tho

Reply:
Yes you're right, wrong comment thread!

> I think this is the same logic as in utils.py branch function, should we abstract it to a function? Not sure if the next two lines can also be included (which would let you include the if statement too).

```suggestion
            target_size = max(first_shape[-1], second_shape[-1])
            
            # Adapt first node if needed
            graph, first_node = adapt_node_shape(graph, first_node, first_shape[-1], target_size)
            
            # Adapt second node if needed
            graph, second_node = adapt_node_shape(graph, second_node, second_shape[-1], target_size)
```

You're right the check would happen twice, we can leave the check in the adapt_node_shape function.

Reply:
There's no need to have the redundant check in core.py. Removed in latest commit.